src/Compilers/Core/VBCSCompiler/ServerDispatcher.cs

```diff
@@ -73,6 +73,30 @@ public static int Main(string[] args)
 
             var pipeName = args[0].Substring(pipeArgPrefix.Length);
 
+            var serverMutexName = $"{pipeName}.server";
+            bool holdsMutex;
+            using (var serverMutex = new Mutex(initiallyOwned: true,
+                                               name: serverMutexName,
+                                               createdNew: out holdsMutex))
+            {
+                if (!holdsMutex)
```

Comment:
Can we add a comment explaining the reasoning for this logic (for the next guy)?